Entities/ACrab.cpp

```diff
@@ -2936,14 +2930,22 @@ void ACrab::DrawHUD(BITMAP *pTargetBitmap, const Vector &targetPos, int whichScr
         }
 
         // Weight and jetpack energy
-        if (m_pJetpack && m_pJetpack->IsAttached() && m_MoveState == JUMP)
-        {
-            float mass = GetMass();
-// TODO: Don't hardcode the mass indicator! Figure out how to calculate the jetpack threshold values
-            str[0] = mass < 135 ? -31 : (mass < 160 ? -30 : -29); str[1] = 0;
-            // Do the blinky blink
-            if ((str[0] == -29 || str[0] == -30) && m_IconBlinkTimer.AlternateSim(250))
-                str[0] = -28;
+        if (m_pJetpack && m_pJetpack->IsAttached() && m_Controller.IsState(BODY_JUMP))
+        {
+			float mass = GetMass();
+			// Draw empty fuel indicator
+			if (m_JetTimeLeft < 100)
+				str[0] = m_IconBlinkTimer.AlternateSim(100) ? -26 : -25;
```

Comment:
It would previously rapidly alter between whatever the default HUD is (ammo etc) and diminished fuel bar, this change simply makes it identical to the `AHuman` variant, which is rather pretty compared to the every-other-frame flashing.